common/utils/src/main/resources/error/error-conditions.json

```
@@ -70,6 +70,12 @@
     ],
     "sqlState" : "42000"
   },
+  "ANSI_DATE_TIME_FIELD_OUT_OF_BOUNDS" : {
+    "message" : [
+      "ANSI mode error: <message>. If necessary set <ansiConfig> to false to bypass this error."
```

This message makes no sense. The problem is not that ANIS mode is on. The problem is that the value is out of bounds.
```suggestion
      "ANSI mode error: <message>. If necessary set <ansiConfig> to false to bypass this error."
```
We should produce a message as good as the existing one. We shoudl NOT just WRAP the existing message.
State the unit: YEAR, MONTH, DAY, HOUR, MINUTE, SECOND
State the range for the unit
State the bad value

---

We are currently catching `java.time.DateTimeException` as `e` and including `e.getMessage` in the error message, but `java.time.DateTimeException` does not provide the ability to obtain each parameter separately.

I just reverted back to the existing error message so can we just use the existing error message for now??

---

A key premise of the error framework is to NOT leak 3. party libraries text and to have control over the language.

---

ANother option is to catch the Java exception and search for the components.
Just need teh few method names and the value. As long as we have tests for each thsi shoudl be safe.

---

> A key premise of the error framework is to NOT leak 3. party libraries text and to have control over the language.

Good point. Let me update the messages

---

I think we can do the similar thing with https://github.com/apache/spark/pull/48260#discussion_r1777216178